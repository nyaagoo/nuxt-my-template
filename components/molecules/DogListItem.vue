<template lang="pug">
  .dog-list-item
    span {{ greet }} {{ dog }}
</template>
<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

@Component({
  name: 'dog-list-item',
  components: {}
})
export default class DogListItem extends Vue {
  // TODO: TypeScriptのParserを入れたら初期化せずに!で置き換える
  @Prop({ required: true }) dog: { name: string; img: string } = {
    name: '1',
    img: '2'
  }
  @Prop({ required: true }) greet: string = 'hey'
}
</script>
<style lang="stylus" scoped>
.dog-list-item
  background #fff
</style>
