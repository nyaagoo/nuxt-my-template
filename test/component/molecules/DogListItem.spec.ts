import { mount } from '@vue/test-utils';
import DogListItem from '~/components/molecules/DogListItem.vue';

describe('DogListItem', () => {
  beforeAll(() => {});
  test('DogListItem.vue is a Vue instance', () => {
    const wrapper = mount(DogListItem, {
      propsData: {
        dogProp: { name: 'test', ruby: 'test' }
      }
    });
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.props().dogProp.name).toBe('test');
  });
});
