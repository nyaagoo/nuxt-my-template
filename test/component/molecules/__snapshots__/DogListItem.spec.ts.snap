// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogListItem Snapshot 1`] = `
<div
  class="dog-list-item"
>
  <span>
    test test2
  </span>
</div>
`;
